datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -255,39 +256,61 @@ where
             if start > to {
                 return None;
             }
-            let end = start + step;
-            debug!(logger, "Requesting logs for blocks [{}, {}]", start, end);
+
+            // This will iterate more than once iff `step` has been reduced,
+            // making as many parallel requests of size `step` as necessary,
+            // respecting `LOG_STREAM_PARALLEL_CHUNKS`.
+            let mut chunk_futures = vec![];
+            let mut offset = start;
+            for _ in 0..*LOG_STREAM_PARALLEL_CHUNKS {
+                if offset == to + 1 {
```

Comment:
Changing this to `> to` would be simpler

Reply:
It would behave the same, but I like how this is more precise.